commit 81151ce462e533551f3284bfdb8e0f461c9220e6
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Wed Jun 1 21:17:34 2022 +0200

    wifi: mac80211: support MLO authentication/association with one link
    
    It might seem a bit pointless to do a multi-link operation
    connection with just a single link, but this is already a
    big change, so for now, limit MLO connections to a single
    link.
    
    Extending that to multiple links will require
     * work on parsing the multi-link element with STA profile
       properly, including element fragmentation;
     * checking the per-link status in the multi-link element
     * implementing logic to have active/inactive links to let
       drivers decide which links should be active;
     * implementing multicast RX deduplication;
     * and likely more.
    
    For now this is still useful since it lets us do multi-link
    connections for the purposes of testing APIs and the higher
    layers such as wpa_supplicant.
    
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/mac80211/tx.c b/net/mac80211/tx.c
index 1d60eab8308a..34dae26646a6 100644
--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -2478,7 +2478,7 @@ int ieee80211_lookup_ra_sta(struct ieee80211_sub_if_data *sdata,
 
 		}
 
-		sta = sta_info_get(sdata, sdata->deflink.u.mgd.bssid);
+		sta = sta_info_get(sdata, sdata->vif.cfg.ap_addr);
 		if (!sta)
 			return -ENOLINK;
 		break;